instruments/tests/test_thorlabs/test_thorlabs_apt.py

```diff
@@ -1174,6 +1174,112 @@ def test_apt_mc_position(init_kdc101):
+        [
+            init_kdc101[0],
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_SET_GENMOVEPARAMS,
+                param1=None,
+                param2=None,
+                dest=0x50,
+                source=0x01,
+                data=struct.pack("<Hl", 0x01, 1000),
+            ).pack(),
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_SET_GENMOVEPARAMS,
+                param1=None,
+                param2=None,
+                dest=0x50,
+                source=0x01,
+                data=struct.pack("<Hl", 0x01, -20000),
+            ).pack(),
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_REQ_GENMOVEPARAMS,
+                param1=0x01,
+                param2=0x00,
+                dest=0x50,
+                source=0x01,
+                data=None,
+            ).pack(),
+        ],
+        [
+            init_kdc101[1],
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_GET_GENMOVEPARAMS,
+                param1=None,
+                param2=None,
+                dest=0x50,
+                source=0x01,
+                data=struct.pack("<Hl", 0x01, -20000),
+            ).pack(),
+        ],
+        sep="",
+    ) as apt:
+        apt.channel[0].backlash_correction = 1000
+        apt.channel[0].backlash_correction = -20000 * u.counts
+        assert apt.channel[0].backlash_correction == u.Quantity(-20000, "counts")
+
+
+def test_apt_mc_backlash_correction_unitful(init_kdc101):
+    """Get / set backlash correction unitful."""
+    with expected_protocol(
+        ik.thorlabs.APTMotorController,
+        [
+            init_kdc101[0],
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_SET_GENMOVEPARAMS,
+                param1=None,
+                param2=None,
+                dest=0x50,
+                source=0x01,
+                data=struct.pack("<Hl", 0x01, 1919),
+            ).pack(),
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_REQ_GENMOVEPARAMS,
+                param1=0x01,
+                param2=0x00,
+                dest=0x50,
+                source=0x01,
+                data=None,
+            ).pack(),
+        ],
+        [
+            init_kdc101[1],
+            ThorLabsPacket(
+                message_id=ThorLabsCommands.MOT_GET_GENMOVEPARAMS,
+                param1=None,
+                param2=None,
+                dest=0x50,
+                source=0x01,
+                data=struct.pack("<Hl", 0x01, 1919),
+            ).pack(),
+        ],
+        sep="",
+    ) as apt:
+        apt.channel[0].motor_model = "PRM1-Z8"
+        corr = 1 * u.deg
+        apt.channel[0].backlash_correction = corr
+
+        corr_received = apt.channel[0].backlash_correction
+        assert corr_received.magnitude == pytest.approx(corr.magnitude, abs=1e-3)
+        assert corr_received.units == corr.units
```

Comment:
In `instruments.testing`, there is the `unit_eq` comparison that would take car of this. However, it is (currently) not possible to pass arguments to `pytest.approx()`. One idea might be to extend `units_eq` to also accept `**kwargs`, which we could pass on to `pytest.approx()`. What do you think?

Reply:
That's a fantastic idea. Shall we do that in a separate PR?

Reply:
I think so, would be cleaner. I can prepare it this afternoon for review, then we can update this PR later on with the new `unit_eq` method.